resources/extruders/MicroSwiss_DirectDrive_extruder_0.def.json

```diff
@@ -0,0 +1,32 @@
+{
+    "version": 2,
+    "name": "Extruder 1",
+    "inherits": "fdmextruder",
+    "metadata": {
+        "_MachineComment": "Ender3_MicroSwiss_DirectDrive",
+		"machine": "creality_base",
```

Comment:
Since the `creality_base` doesn't use these extruders, this causes the automated tests to fail. Similar for extruder 1.

Reply:
This one I'm actually confused about, but thanks for pointing out why the build failed. I wasn't sure why it had failed, since this is part of the printer profile I'm currently using just fine in cura currently. So the file _works_, but understand it may not be _correct_. 
Do you think that the only issue is this line: ` "machine": "creality_base",` specifies creality_base instead of `"name": "Ender-3 MicroSwiss DirectDrive",` (as specified by the printer's .def.json file ?

PS: As with the above comment, I'll remove the extraneous comments from the commits. I'll also convert the spaces in the _name_ printer.def.json file to underscores to match the correct scheme (_creality_ender3_).

Reply:
The tests validate whether the link goes both ways, indeed, which is not required for the printer to be added but may play into other things. I'm actually not sure why it needs that, but the issue arises from the metadata being necessary at all for which there is probably a good reason. Even if it's not used for adding the printer it could be used elsewhere.